### PR TITLE
add visiblity timeout feature

### PR DESCRIPTION
**Pull request by jaesharma**

### CODE REVIEW

File: src/icsqs/main.mo [hunk 1]
```diff
@@ -5,17 +5,20 @@ import Debug "mo:base/Debug";
...
 import UUID "mo:uuid/UUID";
+import Time "mo:base/Time";
 
 shared (install) actor class ICSQS(authorizedPrincipals: List.List<Principal>) =
 this { // Bind the optional `this` argument (any name wi
 
  type Message = {
    id: Text;
    message: Text;
+   last_processed: Int;
```
Comment (ritvick) [inline src/icsqs/main.mo]: lastRead

**Review by ritvick — CHANGES_REQUESTED** [review]

File: src/icsqs/main.mo (hunk 1 above)
Comment (jaesharma) [reply src/icsqs/main.mo]: fixed

File: src/icsqs/main.mo [hunk 2]
```diff
@@ -5,17 +5,20 @@ import Debug "mo:base/Debug";
...
 
  type Message = {
    id: Text;
    message: Text;
+   lastRead: Int;
  };
 
  // list of message objects
  var queueData: List.List<Message>  = List.nil();
+ var visiblit_timeout: Nat = 30; // in seconds
```
Comment (ritvick) [inline src/icsqs/main.mo]: Shouldn't this be stable ?

File: src/icsqs/main.mo [hunk 3]
```diff
@@ -5,17 +5,20 @@ import Debug "mo:base/Debug";
...
 this { // Bind the optional `this` argument (any name wi
 
  type Message = {
    id: Text;
    message: Text;
+   lastRead: Int;
  };
 
  // list of message objects
  var queueData: List.List<Message>  = List.nil();
```
Comment (ritvick) [inline src/icsqs/main.mo]: stable ?

File: src/icsqs/main.mo [hunk 4]
```diff
@@ -90,17 +82,58 @@ this { // Bind the optional `this` argument (any name wi
...
-  let messages: List.List<Message> = List.take<Message>(queueData, count);
-  // remove first `count` number of messages from the list
-  queueData := List.drop<Message>(queueData, count);
-  return messages;
+  var message_count: Int = count;
+  var messages: List.List<Message> = List.nil();
+  let current_time: Int = Time.now();
+
+  queueData := List.map<Message, Message>(queueData, func(message: Message): Message {
+    let last_processed_time: Int = (current_time - message.lastRead) / 1_000_000_000;
```
Comment (ritvick) [inline src/icsqs/main.mo]: lastReadTime ?
Can we be consistant here in terminology.

**Review by ritvick — APPROVED** [review]

**Pull request merged**